File: webapp/src/app/dto/rental-property.dto.ts
import { AccessibilityDTO } from './accessibility.dto';
import { AmenityDTO } from './amenity.dto';

export class RentalPropertyDTO {
  constructor(
    private _id: string,
    private _accessibility: AccessibilityDTO,
    private _amenity: AmenityDTO,
    private _name: string,
    private _description: string,
    private _listed: boolean,
    private _pricePerNight: number,
    private _beds: number,
    private _bedrooms: number,
    private _bathrooms: number,
    private _street: string,
    private _number: string,
    private _postalCode: string
  ) {}

  public get id(): string {
    return this._id;
  }

  public get accessibility(): AccessibilityDTO {
    return this._accessibility;
  }

  public set accessibility(accessibility: AccessibilityDTO) {
    this._accessibility = accessibility;
  }

  public get amenity(): AmenityDTO {
    return this._amenity;
  }

  public set amenity(amenity: AmenityDTO) {
    this._amenity = amenity;
  }

  public get name(): string {
    return this._name;
  }

  public set name(name: string) {
    this._name = name;
  }

  public get description(): string {
    return this._description;
  }

  public set description(description: string) {
    this._description = description;
  }

  public get listed(): boolean {
    return this._listed;
  }

  public set listed(listed: boolean) {
    this._listed = listed;
  }

  public get pricePerNight(): number {
    return this._pricePerNight;
  }

  public set pricePerNight(pricePerNight: number) {
    this._pricePerNight = pricePerNight;
  }

  public get beds(): number {
    return this._beds;
  }

  public set beds(beds: number) {
    this._beds = beds;
  }

  public get bedrooms(): number {
    return this._bedrooms;
  }

  public set bedrooms(bedrooms: number) {
    this._bedrooms = bedrooms;
  }

  public get bathrooms(): number {
    return this._bathrooms;
  }

  public set bathrooms(bathrooms: number) {
    this._bathrooms = bathrooms;
  }

  public get street(): string {
    return this._street;
  }

  public set street(street: string) {
    this._street = street;
  }

  public get number(): string {
    return this._number;
  }

  public set number(number: string) {
    this._number = number;
  }

  public get postalCode(): string {
    return this._postalCode;
  }

  public set postalCode(postalCode: string) {
    this._postalCode = postalCode;
  }
}

File: webapi/src/main/java/dev/hekmyr/olids/api/dto/RentalPropertyDTO.java
package dev.hekmyr.olids.api.dto;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import dev.hekmyr.olids.api.entity.RentalProperty;
import java.util.UUID;

public class RentalPropertyDTO {

  private UUID id;

  @JsonSerialize
  private AccessibilityDTO accessibility;

  @JsonSerialize
  private AmenityDTO amenity;

  private String name;
  private String description;
  private boolean listed;
  private int pricePerNight;
  private int beds;
  private int bedrooms;
  private int bathrooms;
  private String street;
  private String number;
  private String postalCode;
  private String image;
  private Integer maxGuests;

  public RentalPropertyDTO() {}

  public RentalPropertyDTO(RentalProperty entity) {
    this.id = entity.getId();
    this.name = entity.getName();
    this.description = entity.getDescription();
    this.listed = entity.isListed();
    this.pricePerNight = entity.getPricePerNight();
    this.beds = entity.getBeds();
    this.bedrooms = entity.getBedrooms();
    this.bathrooms = entity.getBathrooms();
    this.street = entity.getStreet();
    this.number = entity.getNumber();
    this.postalCode = entity.getPostalCode();
    this.image = entity.getImage();
    this.maxGuests = entity.getMaxGuests();
    this.amenity = new AmenityDTO(entity.getAmenity());
    this.accessibility = new AccessibilityDTO(entity.getAccessibility());
  }

  public UUID getId() {
    return id;
  }

  public void setId(UUID id) {
    this.id = id;
  }

  public AccessibilityDTO getAccessibility() {
    return accessibility;
  }

  public void setAccessibility(AccessibilityDTO accessibility) {
    this.accessibility = accessibility;
  }

  public AmenityDTO getAmenity() {
    return amenity;
  }

  public void setAmenity(AmenityDTO amenity) {
    this.amenity = amenity;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public Boolean isListed() {
    return listed;
  }

  public void setListed(Boolean listed) {
    this.listed = listed;
  }

  public Integer getPricePerNight() {
    return pricePerNight;
  }

  public void setPricePerNight(Integer pricePerNight) {
    this.pricePerNight = pricePerNight;
  }

  public Integer getBeds() {
    return beds;
  }

  public void setBeds(Integer beds) {
    this.beds = beds;
  }

  public Integer getBedrooms() {
    return bedrooms;
  }

  public void setBedrooms(Integer bedrooms) {
    this.bedrooms = bedrooms;
  }

  public Integer getBathrooms() {
    return bathrooms;
  }

  public void setBathrooms(Integer bathrooms) {
    this.bathrooms = bathrooms;
  }

  public String getStreet() {
    return street;
  }

  public void setStreet(String street) {
    this.street = street;
  }

  public String getNumber() {
    return number;
  }

  public void setNumber(String number) {
    this.number = number;
  }

  public String getPostalCode() {
    return postalCode;
  }

  public void setPostalCode(String postalCode) {
    this.postalCode = postalCode;
  }

  public String getImage() {
    return image;
  }

  public void setImage(String image) {
    this.image = image;
  }

  public int getMaxGuests() {
    return maxGuests;
  }

  public void setMaxGuests(int maxGuests) {
    this.maxGuests = maxGuests;
  }  
}
